lib/internal/webstorage.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const {
+  MathAbs,
+  ObjectDefineProperty,
+  StringPrototypeCharCodeAt,
+} = primordials;
+const { emitExperimentalWarning } = require('internal/util');
+const { kConstructorKey, Storage } = internalBinding('webstorage');
+const { basename, join } = require('path');
+
+emitExperimentalWarning('Web storage');
+
+module.exports = { Storage };
+
+let lazyLocalStorage;
+let lazySessionStorage;
+
+ObjectDefineProperty(module.exports, 'localStorage', {
+  __proto__: null,
+  configurable: true,
+  enumerable: true,
+  get() {
+    if (lazyLocalStorage === undefined) {
+      const entry = process.argv[1] ?? process.argv[0];
+      const base = basename(entry);
+      const hash = hashCode(entry);
+      const location = join(process.cwd(), `${base}.${hash}.localstorage`);
```

Comment:
I think we would need a CLI flag to set this, and possibly an API to create new localStorage instances

Reply:
+1 on the CLI flag for the `location`. The idea of an API to create new instances is good but let's defer that to a follow on PR.